blue_only.py

```diff
@@ -1,221 +1,290 @@
+    """
+    Checks if the shape is close to circular.
+
+    Return: True is the shape is circular, false if it is not.
+    """
+    perimeter = cv2.arcLength(contour, True)
+
+    # Check if the perimeter is zero
+    if perimeter == 0.0:
+        return False
+
+    area = cv2.contourArea(contour)
+    circularity = 4 * np.pi * (area / (perimeter * perimeter))
+    return circularity > CONTOUR_MINIMUM
+
+
+def is_contour_large_enough(contour: np.ndarray, min_diameter: float) -> bool:
+    """
+    Checks if the shape is larger than the provided diameter.
+
+    Return: True if it is, false if it not.
+    """
+    _, radius = cv2.minEnclosingCircle(contour)
+    diameter = radius * 2
+    return diameter >= min_diameter
+
+
+def calc_target_distance(height_agl: float, x_rad: float, y_rad: float) -> float:
+    """
+    Get the horizontal distance TODO
+    """
+    x_ground_dist_m = math.tan(x_rad) * height_agl
+    y_ground_dist_m = math.tan(y_rad) * height_agl
+    ground_hyp = math.sqrt(math.pow(x_ground_dist_m, 2) + math.pow(y_ground_dist_m, 2))
+    print("Required horizontal correction (m): ", ground_hyp)
+    target_to_vehicle_dist = math.sqrt(math.pow(ground_hyp, 2) + math.pow(height_agl, 2))
+    print("Distance from vehicle to target (m): ", target_to_vehicle_dist)
+    return target_to_vehicle_dist
+
+
+# Callback expects `self` as first argument
+# pylint: disable-next=unused-argument
+def my_allow_unsigned_callback(self: object, message_id: int) -> bool:
+    """
+    Specify which messages to accept.
+    """
+    # Allow radio status messages
+    return message_id == pymavlink.mavutil.mavlink.MAVLINK_MSG_ID_RADIO_STATUS
+
+
+def detect_landing_pads_contour(image: np.ndarray) -> "None | tuple":
+    """
+    Detect landing pads using contours.
+
+    image: Current image frame
+    """
+    kernel = np.ones((2, 2), np.uint8)
+
+    gray_image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    threshold = 180
+    im_bw = cv2.threshold(gray_image, threshold, 255, cv2.THRESH_BINARY)[1]
+    im_dilation = cv2.dilate(im_bw, kernel, iterations=1)
+    contours, hierarchy = cv2.findContours(im_dilation, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+
+    if len(contours) == 0:
+        return None
+
+    contours_with_children = set(i for i, hier in enumerate(hierarchy[0]) if hier[2] != -1)
+    parent_circular_contours = [
+        cnt
+        for i, cnt in enumerate(contours)
+        if is_contour_circular(cnt)
+        and is_contour_large_enough(cnt, 7)
+        and i in contours_with_children
+    ]
+    contour_image = copy.deepcopy(image)
+    cv2.drawContours(contour_image, parent_circular_contours, -1, (0, 255, 0), 2)
+    largest_contour = max(parent_circular_contours, key=cv2.contourArea, default=None)
+
+    if not largest_contour:
+        return None
+
+    return tuple(cv2.boundingRect(largest_contour))
+
+
+def detect_landing_pads_yolo(image: np.ndarray, config: dict) -> "None | tuple":
```

Comment:
same as above, let's change the function to return `False, None` and `True, output`